roachprod-stress: works incorrectly with -stderr=false

```
make roachprod-stress TESTS=TestStoreRangeWaitForApplication PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-stderr=true -maxtime 20m -timeout 10m' CLUSTER='tobias-stress'
```

will output something like the below (not particularly helpful, since the error is missing).
I would just deregister the flag or force it to be set to true.

This is made relevant by the fact that our standard test failure issue make stress invocation specifies `stderr=false`.

```
go test -i -v -c -o storage.test -tags ' make x86_64_pc_linux_gnu' -ldflags '-X github.com/cockroachdb/cockroach/pkg/build.typ=development -extldflags "" -X "github.com/cockroachdb/cockroach/pkg/build.tag=v2.2.0-alpha.00000000-1245-gb6904f9" -X "github.com/cockroachdb/cockroach/pkg/build.rev=b6904f97044f3540f97ad8830e150955b082735e" -X "github.com/cockroachdb/cockroach/pkg/build.cgoTargetTriple=x86_64-pc-linux-gnu"  ' -run "."  -timeout 8m github.com/cockroachdb/cockroach/pkg/storage
tobias-stress: putting (dist) bin.docker_amd64/stress stress
   1: done
   2: done
   3: done
   4: done
   5: done
   6: done
   7: done
   8: done
   9: done
  10: done
tobias-stress: putting (dist) ./storage.test storage.test
   1: done
   2: done
   3: done
   4: done
   5: done
   6: done
   7: done
   8: done
   9: done
  10: done
0 runs so far, 0 failures, over 5s
552 runs so far, 0 failures, over 10s
1133 runs so far, 0 failures, over 15s
1715 runs so far, 0 failures, over 20s
2298 runs so far, 0 failures, over 25s
2883 runs so far, 0 failures, over 30s
3472 runs so far, 0 failures, over 35s
4046 runs so far, 0 failures, over 40s
4630 runs so far, 0 failures, over 45s
5221 runs so far, 0 failures, over 50s
5796 runs so far, 0 failures, over 55s
6376 runs so far, 0 failures, over 1m0s
6948 runs so far, 0 failures, over 1m5s
7528 runs so far, 0 failures, over 1m10s
8110 runs so far, 0 failures, over 1m15s
8696 runs so far, 0 failures, over 1m20s
9273 runs so far, 0 failures, over 1m25s
9857 runs so far, 0 failures, over 1m30s
10440 runs so far, 0 failures, over 1m35s
11017 runs so far, 0 failures, over 1m40s
11601 runs so far, 0 failures, over 1m45s
12178 runs so far, 0 failures, over 1m50s
12753 runs so far, 0 failures, over 1m55s
13337 runs so far, 0 failures, over 2m0s
13919 runs so far, 0 failures, over 2m5s
14497 runs so far, 0 failures, over 2m10s
15077 runs so far, 0 failures, over 2m15s
15662 runs so far, 0 failures, over 2m20s
16238 runs so far, 0 failures, over 2m25s
16826 runs so far, 0 failures, over 2m30s
17407 runs so far, 0 failures, over 2m35s
17982 runs so far, 0 failures, over 2m40s
18572 runs so far, 0 failures, over 2m45s
19160 runs so far, 0 failures, over 2m50s
19742 runs so far, 0 failures, over 2m55s
20320 runs so far, 0 failures, over 3m0s
20903 runs so far, 0 failures, over 3m5s
21491 runs so far, 0 failures, over 3m10s
22070 runs so far, 0 failures, over 3m15s
22650 runs so far, 0 failures, over 3m20s
23233 runs so far, 0 failures, over 3m25s
23817 runs so far, 0 failures, over 3m30s
24402 runs so far, 0 failures, over 3m35s
24987 runs so far, 0 failures, over 3m40s
25570 runs so far, 0 failures, over 3m45s
26159 runs so far, 0 failures, over 3m50s
26739 runs so far, 0 failures, over 3m55s
27322 runs so far, 0 failures, over 4m0s
27906 runs so far, 0 failures, over 4m5s
28490 runs so far, 0 failures, over 4m10s
29074 runs so far, 0 failures, over 4m15s

context canceled
Error:  exit status 1

shutting down
29074 runs completed, 0 failures, over 4m16s
context canceled
FAIL
make: *** [roachprod-stress] Error 1
```

--- a/pkg/cmd/roachprod-stress/main.go
+++ b/pkg/cmd/roachprod-stress/main.go
@@ -48,7 +48,7 @@ var (
 	flagMaxTime = flags.Duration("maxtime", 0, "maximum time to run")
 	flagMaxRuns = flags.Int("maxruns", 0, "maximum number of runs")
 	_           = flags.Int("maxfails", 1, "maximum number of failures")
-	_           = flags.Bool("stderr", true, "output failures to STDERR instead of to a temp file")
+	flagStderr  = flags.Bool("stderr", true, "output failures to STDERR instead of to a temp file")
 )
 
 func roundToSeconds(d time.Duration) time.Duration {
@@ -71,6 +71,10 @@ func run() error {
 	cluster := os.Args[1]
 	if err := flags.Parse(os.Args[2:]); err != nil {
 		return err
+	}
+
+	if !*flagStderr {
+		return errors.New("-stderr=false is unsupported, please tee to a file (or implement the feature)")
 	}
 
 	if *flagP <= 0 || *flagTimeout < 0 || len(flags.Args()) == 0 {

